initfiles/bash/etc/init.d/hpcc_common.in

```diff
@@ -559,26 +559,24 @@ stop_component() {
 
     eval $stopcmd
 
+    unlock ${LOCKPATH}
+    removePid ${PIDPATH}
+
     RESULT=0
-    local counter=0
-    while [[ $RESULT -eq 0 && $counter -le 20 ]]; do
+    local waittime=30
+    while [[ $RESULT -ne 1 && $waittime -gt 0 ]]; do
         check_status ${PIDPATH} ${LOCKPATH} ${COMPPIDPATH} 0
         RESULT=$?
-        ((counter++))
-        if [[ $RESULT -eq 0 ]]; then
-            sleep 1
-        fi
+        ((waittime--))
+        sleep 1
```

Comment:
If it does end up waiting up to the limit.. might be good it if it logged some feedback every so often in this loop